src/test/java/com/google/ehub/servlets/ProfileServletTest.java

```diff
@@ -0,0 +1,194 @@
+    when(request.getParameter(NAME_PROPERTY_KEY)).thenReturn(null);
+    when(request.getParameter(EMAIL_PROPERTY_KEY)).thenReturn(null);
+    when(request.getParameter(BIO_PROPERTY_KEY)).thenReturn(null);
+    when(request.getParameter(EDIT_PROPERTY_KEY)).thenReturn(null);
+    when(request.getParameter(USERNAME_PROPERTY_KEY)).thenReturn(null);
+
+    servlet.doPost(request, response);
+
+    verify(response).sendError(HttpServletResponse.SC_BAD_GATEWAY, POST_ERROR_MESSAGE);
+  }
+
+  @Test
+  public void postRequestWithEmptyParams_ErrorGetsSent() throws IOException {
+    when(request.getParameter(NAME_PROPERTY_KEY)).thenReturn("");
+    when(request.getParameter(EMAIL_PROPERTY_KEY)).thenReturn("");
+    when(request.getParameter(BIO_PROPERTY_KEY)).thenReturn("");
+    when(request.getParameter(EDIT_PROPERTY_KEY)).thenReturn("");
+    when(request.getParameter(USERNAME_PROPERTY_KEY)).thenReturn("");
+
+    servlet.doPost(request, response);
+
+    verify(response).sendError(HttpServletResponse.SC_BAD_GATEWAY, POST_ERROR_MESSAGE);
+  }
+
+  @Test
+  public void postRequestWithValidParams_SendsRedirect() throws IOException {
+    Entity userEntity = new Entity(PROFILE_ITEM_KIND);
+
+    userEntity.setProperty(NAME_PROPERTY_KEY, NAME);
+    userEntity.setProperty(EMAIL_PROPERTY_KEY, EMAIL);
+    userEntity.setProperty(USERNAME_PROPERTY_KEY, USERNAME);
+    userEntity.setProperty(BIO_PROPERTY_KEY, BIO);
+
+    datastoreService.put(userEntity);
+
+    when(request.getParameter(NAME_PROPERTY_KEY)).thenReturn(NAME);
+    when(request.getParameter(EMAIL_PROPERTY_KEY)).thenReturn(EMAIL);
+    when(request.getParameter(BIO_PROPERTY_KEY)).thenReturn(BIO);
+    when(request.getParameter(USERNAME_PROPERTY_KEY)).thenReturn(USERNAME);
+
+    servlet.doPost(request, response);
+
+    verify(response).sendRedirect(REDIRECT);
+  }
+
+  @Test
+  public void postResquestWithFalseEditParam_SendsRedirect() throws IOException {
+    Entity userEntity = new Entity(PROFILE_ITEM_KIND);
+
+    userEntity.setProperty(NAME_PROPERTY_KEY, NAME);
+    userEntity.setProperty(EMAIL_PROPERTY_KEY, EMAIL);
+    userEntity.setProperty(USERNAME_PROPERTY_KEY, USERNAME);
+    userEntity.setProperty(BIO_PROPERTY_KEY, BIO);
+
+    datastoreService.put(userEntity);
+
+    when(request.getParameter(NAME_PROPERTY_KEY)).thenReturn(NAME);
+    when(request.getParameter(EMAIL_PROPERTY_KEY)).thenReturn(EMAIL);
+    when(request.getParameter(BIO_PROPERTY_KEY)).thenReturn(BIO);
+    when(request.getParameter(USERNAME_PROPERTY_KEY)).thenReturn(USERNAME);
+    when(request.getParameter(EDIT_PROPERTY_KEY)).thenReturn(FALSE_EDIT_VALUE);
+
+    servlet.doPost(request, response);
+
+    verify(response).sendRedirect(REDIRECT);
+  }
+
+  @Test
+  public void getRequestWithLoggedOutUser_ErrorGetsSent() throws IOException {
+    helper.setEnvIsLoggedIn(false);
+
+    servlet.doGet(request, response);
+
+    verify(response).sendError(HttpServletResponse.SC_BAD_GATEWAY, ERROR_MESSAGE);
```

Comment:
Since the error message is arbitrary I think it would be better if you just ignore the parameter and instead declare it as anyString() using Mockito. an example for this code would look like this: ```verify(response).sendError(eq(HttpServletResponse.SC_BAD_REQUEST), anyString());```
Checkout Mockito any https://www.journaldev.com/21876/mockito-argument-matchers-any-eq

Reply:
This isn't working for me. I tried importing the class as well, but it won't work

Reply:
Can you try static import:`import static org.mockito.ArgumentMatchers.anyString;`